### DISCUSSION TITLE
How to backup?

### DISCUSSION THREAD

**Discussion by bertmelis** [immich-app/immich#741] [Q&A]
What is currently the best way to reliably create backups?

I now stop the entire Immich stack and backup the volumes and database. (Database is separate and keeps online but I could also backup the db volume). This however isn't always reliable because I use Podman instead of Docker and the backup user can only manage other users' containers in some hacky way.

I could also only stop the proxy container but I understand there are some things running in the background. I have no clue whether or not these tasks could potentially render the backup unusable.

- So, how do you create backups?
- Is it safe to just backup a running instance and will Immich be able to handle a recovery?

**Comment by nep2ner** [comment]
I was thinking about using rclone/rsync to backup the database and volumes. Does Immich need stopped for this? I would think not?

**Reply by bo0tzz** [reply]
You should be aware that rclone/rsync are not backup tools. I recommend taking a look at [this list](https://github.com/awesome-foss/awesome-sysadmin#backups) of backup software.

**Reply by bondeabhijeet** [reply]
They have added rclone to their backup list now.

**Comment by bo0tzz** [comment]
The postgres database will keep its files open as long as it is running. A backup taken of those files -might- restore properly (postgres is pretty resilient), but to be certain you really should stop the database container. So, simply said:
1. Stop the stack (including database)
2. Point your usual backup tool at the library and database volumes
3. Done :)
